drones/image_processing/normalization.py

```diff
@@ -0,0 +1,250 @@
+import numpy as np
+import cv2 as cv
+
+
+def min_max_norm(image: np.ndarray) -> np.ndarray:
+    """Apply min max norm on given image. It should be used with OpenCv
+
+    Parameters:
+    ----------
+    image : np.ndarray
+        image to be normalized
+
+    Returns:
+    ----------
+    image : np.ndarray
+        normalized version of input image
+    """
+    norm_image = np.zeros((image.shape[0], image.shape[1]))
+    return cv.normalize(image, norm_image, 0, 255, cv.NORM_MINMAX)
+
+
+def zero_one_norm(image: np.ndarray) -> np.ndarray:
+    """Apply zero one pixel range. It should be used with AI
+
+    Parameters:
+    ----------
+    image : np.ndarray
+        image to be normalized
+
+    Returns:
+    ----------
+    image : np.ndarray
+        normalized version of input image
+    """
+    norm_image = np.zeros((image.shape[0], image.shape[1]))
+    return cv.normalize(image, norm_image, 0, 1, cv.NORM_MINMAX)
+
+
+def gray_norm(image: np.ndarray) -> np.ndarray:
+    """Convert image to gray image
+
+    Parameters:
+    ----------
+    image : np.ndarray
+        image to be changed into gray
+
+    Returns:
+    ----------
+    image : np.ndarray
+        gray image
+    """
+    return cv.cvtColor(image, cv.COLOR_BGR2GRAY)
+
+
+def histogram_equalization_gray(image: np.ndarray) -> np.ndarray:
+    """Apply histogram normalization on single channel. If image have more channels, it will return input
+
+    Parameters:
+    ----------
+    image : np.ndarray
+        image to be normalized(grayscale)
+
+    Returns:
+    ----------
+    image : np.ndarray
+        normalized version of input image
+    """
+    if len(image.shape) > 2:
+        return image
```

Comment:
Suggest to inform the user about returning unprocessed input instead of histogram.